utils/generic.py

```diff
@@ -878,4 +932,152 @@ def assign_population_pcs(
     pop_pc_pd.loc[probs['max'] < min_prob, output_col] = missing_label
     pop_pc_pd = pop_pc_pd.drop(pc_cols, axis='columns')
 
-    return pop_pc_pd, pop_clf
+    logger.info("Found the following sample count after population assignment: {}".format(
+        ", ".join(f'{pop}: {count}' for pop, count in Counter(pop_pc_pd[output_col]).items())
+    ))
+
+    if hail_input:
+        pops_ht = hl.Table.from_pandas(pop_pc_pd, key=list(pop_pca_scores.key))
+        pops_ht.annotate_globals(
+            assign_pops_from_pc_params=hl.struct(
+                min_assignment_prob=min_prob
+            )
+        ).persist()
```

Comment:
you should persist when you're going to use a result multiple times to avoid recomputing it. In this case, I think this is a remain from copy/pasting it from my own script as their is no clear reason to have it here. I'll remove it!

Reply:
Thank you for explaining it!!